batavia/VirtualMachine.js

```diff
@@ -1716,16 +1720,43 @@ VirtualMachine.prototype.byte_POP_EXCEPT = function() {
     this.unwind_block(block)
 }
 
-// VirtualMachine.prototype.byte_SETUP_WITH = function(dest) {
-//         ctxmgr = this.pop()
-//         this.push(ctxmgr.__exit__)
-//         ctxmgr_obj = ctxmgr.__enter__()
-//         if PY2:
-//             this.push_block('with', dest)
-//         elif PY3:
-//             this.push_block('finally', dest)
-//         this.push(ctxmgr_obj)
-// }
+VirtualMachine.prototype.byte_SETUP_WITH = function(dest) {
+    var mgr = this.top()
```

Comment:
Agreed. The state of the stack is an internal detail; and in this particular case, it's a transient detail - so go right ahead.